lib/nerves_runtime/log_tailer.ex

```diff
@@ -0,0 +1,137 @@
+defmodule Nerves.Runtime.LogTailer do
+  use GenServer
+
+  require Logger
+
+  defp gen_server_name(:syslog), do: __MODULE__.Syslog
```

Comment:
This feels like an unnecessary redundancy here.  You are already uniquely identifying the process in the `worker` spec by ID, there's no reason why you can't use those atoms for the GenServer name too.

Reply:
In other words, use `Nerves.Runtime.LogTailer.Syslog` or `Nerves.Runtime.LogTailer.Kmsg` as both the name and ID.